src/components/editor/Editor.tsx

```diff
@@ -1,71 +1,101 @@
-import React from "react";
-import AttributeEditor from "./attributeEditor/AttributeEditor";
-import ResourceProfileMapping from "./resourceProfileMapping/ResourceProfileMapping";
-import StructuredefSettings from "../structuredefSettings/StructuredefSettings";
+import React, { useState } from "react";
 import { useSelector } from "react-redux";
-import { RootState } from "../../state/store";
-import {
-  IElementDefinition,
-  IStructureDefinition
-} from "@ahryman40k/ts-fhir-types/lib/R4";
 
-const Editor: React.FC<{}> = () => {
-  const { loading, structureDefinition, selectedAttributeId } = useSelector(
-    (state: RootState) => state.resource
-  );
-  const structuredefSettings: string = "structuredefSettings";
-  let attribute: IElementDefinition | undefined = undefined;
+import clsx from "clsx";
+import { IElementDefinition } from "@ahryman40k/ts-fhir-types/lib/R4";
+import { Paper, Container, Typography, Breadcrumbs } from "@material-ui/core";
 
-  if (selectedAttributeId && selectedAttributeId !== structuredefSettings) {
-    attribute = structureDefinition?.snapshot?.element.find(
-      (attribute: IElementDefinition) => attribute.id === selectedAttributeId
-    );
-  }
+import AttributeEditor from "components/editor/attributeEditor/AttributeEditor";
+import { ButtonDownloadYouyou } from "components/utils/utils";
+import Navbar from "components/navbar/Navbar";
+import ResourceProfileMapping from "components/editor/resourceProfileMapping/ResourceProfileMapping";
+import StructureDefSettings from "components/structureDefSettings/StructureDefSettings";
+import { RootState } from "state/store";
 
-  const renderAttributeEditor = (structureDef: IStructureDefinition) => {
-    if (selectedAttributeId === structuredefSettings) {
-      return (
-        structureDefinition && (
-          <StructuredefSettings
-            structureDefinition={structureDefinition}
-            type="resource"
-          />
-        )
-      );
-    } else {
-      return (
-        <AttributeEditor
-          attribute={attribute}
-          structureDefinition={structureDef}
-        />
-      );
-    }
-  };
+import useStyles from "components/editor/style";
+
+const Editor: React.FC<{}> = () => {
+  const {
+    loading,
+    structureDefinition,
+    selectedAttributeId,
+    selectStructureDefMeta
+  } = useSelector((state: RootState) => state.resource);
+  const classes = useStyles();
+  const splitedAttributeSelected = selectedAttributeId?.split(".");
+  const [attribute] = useState(
+    structureDefinition?.snapshot?.element.find(
+      (attribute: IElementDefinition) => attribute.id === selectedAttributeId
+    )
+  );
```

Comment:
I am not sure using 'useState' to get the matching attribute is a good idea because you don't even instantiate the setAttribute method. 
Basically, 'attribute' depends on 'selectedAttributeId' and 'structureDefinition' variables you get from the store. So, it is possible to create it from within the 'useSelector' hook and return it.